### PR TITLE
Bug fix/graph node keep all flags on clone

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

Fixed a situation where GraphNodes during clone operation drops the isSmart and isDisjoint values.
This could show up on more complex queries if there are alternative plans to evaluate.
This has the side-effect that a plan could be selected that does not make use of the Sharding Information.

This situations can only be triggered by Enterprise version.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [X] Backports required for: (3.8 maybe earlier tbd.)

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [X] GitHub issue / Jira ticket number: DEVSUP-838
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [X] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [X] This PR adds tests that were used to verify all changes:
  - [X] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: arangod/Aql/GraphNode.h [hunk 1]
```diff
@@ -197,9 +197,23 @@ class GraphNode : public ExecutionNode {
 
   graph::Graph const* graph() const noexcept;
 
+#ifdef ARANGODB_USE_GOOGLE_TESTS
+  // Internal Helpers used in tests to modify enterprise detections
+  // These should not be used in Production, as their detection
+  // is implemented in constructors.
```
Comment (hkernbach) [inline arangod/Aql/GraphNode.h:203]:
```suggestion
#ifdef ARANGODB_USE_GOOGLE_TESTS
  // Internal helpers used in tests to modify enterprise detections.
  // These should not be used in production, as their detection
  // is implemented in constructors.
```

**Review by jsteemann — APPROVED** [review]
LGTM!

**Review by hkernbach — APPROVED** [review]
LGTM. Only very minor lang. improvement mentioned.

**Comment by mchacki** [comment]
The Jenkins has shown an unrelated issue, which is most likely due to some hw configuration.
Will be investigated separately.

**Pull request merged**